torch/csrc/dynamo/eval_frame.c

```diff
@@ -611,6 +611,12 @@ static PyObject* _custom_eval_frame(
 
   PyObject* backend = get_backend(callback);
 
+
+  // We don't run the current custom_eval_frame behavior for guards.
+  // So we temporarily set the callback to Py_None to drive the correct behavior
+  // in the shim.
+  eval_frame_callback_set(Py_None);
```

Comment:
This ensure that for torch._dynamo.run mode or `extra_state_cache_limit_hit(extra)` also has callback disabled (condition at L624)

Reply:
Yes.